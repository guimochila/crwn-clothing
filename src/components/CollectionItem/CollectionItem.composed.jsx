import React from 'react';
import { Mutation } from 'react-apollo';

import CollectionItem from './CollectionItem';
import { ADD_ITEM_TO_CART } from '../../graphql/queries';

const CollectionItemComposed = props => (
  <Mutation mutation={ADD_ITEM_TO_CART} variables={{ item: props.item }}>
    {addItemToCart => (
      <CollectionItem {...props} addItem={item => addItemToCart(item)} />
    )}
  </Mutation>
);

export default CollectionItemComposed;
