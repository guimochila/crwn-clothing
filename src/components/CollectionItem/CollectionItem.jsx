import React from 'react';
import {
  CollectionItemContainer,
  CollectionFooterContainer,
  AddButton,
  BackgroundImage,
  NameContainer,
  PriceContainer,
} from './CollectionItem.styles';

function CollectionItem({ item, addItem }) {
  const { name, price, imageUrl } = item;

  return (
    <CollectionItemContainer>
      <BackgroundImage className="image" imageUrl={imageUrl} />
      <CollectionFooterContainer>
        <NameContainer>{name}</NameContainer>
        <PriceContainer>{price}</PriceContainer>
      </CollectionFooterContainer>
      <AddButton inverted onClick={() => addItem(item)}>
        Add to cart
      </AddButton>
    </CollectionItemContainer>
  );
}

export default CollectionItem;
