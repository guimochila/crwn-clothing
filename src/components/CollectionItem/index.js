import CollectionItemComposed from './CollectionItem.composed';

export default CollectionItemComposed;
