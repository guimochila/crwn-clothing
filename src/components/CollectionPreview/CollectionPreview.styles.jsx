import styled from 'styled-components';
import { Link } from 'react-router-dom';

export const CollectionPreviewContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
`;

export const TitleLink = styled(Link)`
  display: inline-block;
  width: 100px;

  h1 {
    font-size: 28px;
    margin-bottom: 25px;
  }
`;

export const PreviewContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;
