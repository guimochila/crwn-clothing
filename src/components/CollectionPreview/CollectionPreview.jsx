import React from 'react';
import { withRouter } from 'react-router-dom';

import {
  CollectionPreviewContainer,
  TitleLink,
  PreviewContainer,
} from './CollectionPreview.styles';
import CollectionItem from '../CollectionItem';

export function CollectionPreview({ title, items, match }) {
  return (
    <CollectionPreviewContainer>
      <TitleLink to={`${match.url}/${title.toLowerCase()}`}>
        <h1>{title.toUpperCase()}</h1>
      </TitleLink>
      <PreviewContainer>
        {items
          .filter((_, index) => index < 4)
          .map(item => (
            <CollectionItem key={item.id} item={item} />
          ))}
      </PreviewContainer>
    </CollectionPreviewContainer>
  );
}

export default withRouter(CollectionPreview);
