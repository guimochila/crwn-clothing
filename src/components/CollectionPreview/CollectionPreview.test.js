import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './CollectionPreview';
import CollectionItem from '../CollectionItem';

describe('CollectionPreview component', () => {
  let wrapper;
  const items = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }];

  const mockProps = {
    match: {
      url: 'https://crwn-clothing.com/items',
    },
    title: 'Hats',
    items,
  };

  beforeEach(() => {
    wrapper = shallow(<CollectionPreview {...mockProps} />);
  });

  test('should render CollectionPreview', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should PreviewContainer can not have more than 4 CollectionItem rendered', () => {
    expect(wrapper.find(CollectionItem).length).toBe(4);
  });

  test('should render the correct URL in TitleLink', () => {
    expect(wrapper.find('TitleLink').prop('to')).toBe(
      `${mockProps.match.url}/${mockProps.title.toLocaleLowerCase()}`,
    );
  });
});
