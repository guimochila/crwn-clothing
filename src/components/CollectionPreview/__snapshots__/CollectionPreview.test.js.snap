// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component should render CollectionPreview 1`] = `
<styled.div>
  <TitleLink
    to="https://crwn-clothing.com/items/hats"
  >
    <h1>
      HATS
    </h1>
  </TitleLink>
  <PreviewContainer>
    <Connect(CollectionItem)
      item={
        Object {
          "id": 1,
        }
      }
      key="1"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 2,
        }
      }
      key="2"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 3,
        }
      }
      key="3"
    />
    <Connect(CollectionItem)
      item={
        Object {
          "id": 4,
        }
      }
      key="4"
    />
  </PreviewContainer>
</styled.div>
`;
